src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -49,6 +60,57 @@ internal bool AutomaticRedirection
                 _automaticRedirection = value;
             }
         }
+        internal bool UseProxy
+        {
+            get
+            {
+                if (_proxyPolicy == CurlProxyUsePolicy.UseDefaultProxy)
+                {
+                    return false;
+                }
+                return _proxyPolicy != CurlProxyUsePolicy.DoNotUseProxy;
```

Comment:
>I'll change the value of UseProxy to return true when the state is defaultProxy. 

That is not right.  If you are returning the value of the `HttpClient.UseProxy` property to the caller, it should be TRUE 2 different values of your policy (defaultproxy, customproxy).  It should be FALSE only when your policy enum is DoNotUseProxy.

The `HttpClientHandler` properties for `UseProxy` and `Proxy` do not reflect any value of your environment variable. They are set by the caller.